pkg/core/location/location.go

```diff
@@ -60,3 +60,65 @@ func (a api) List(ctx context.Context, page, limit int, search string) ([]Locati
 
 	return responsePayload.Data.Data, nil
 }
+
+func (a api) Get(ctx context.Context, identifier string) (Location, error) {
+	url := fmt.Sprintf(
+		"%s%s.json/%s",
+		a.client.BaseURL(),
+		pathPrefix,
+		identifier,
+	)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return Location{}, fmt.Errorf("could not create location get request: %w", err)
+	}
+
+	httpResponse, err := a.client.Do(req)
+	if err != nil {
+		return Location{}, fmt.Errorf("could not execute location get request: %w", err)
+	}
+	if httpResponse.StatusCode >= 500 && httpResponse.StatusCode < 600 {
+		return Location{}, fmt.Errorf("could not execute location get request, got response %s", httpResponse.Status)
+	}
+
+	var location Location
+	err = json.NewDecoder(httpResponse.Body).Decode(&location)
+	_ = httpResponse.Body.Close()
+	if err != nil {
+		return Location{}, fmt.Errorf("could not decode location get response: %w", err)
+	}
+
+	return location, nil
+}
+
+func (a api) GetByCode(ctx context.Context, code string) (Location, error) {
+	url := fmt.Sprintf(
+		"%s%s/by-code.json/%s",
+		a.client.BaseURL(),
+		pathPrefix,
+		code,
+	)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return Location{}, fmt.Errorf("could not create location get-by-code request: %w", err)
+	}
+
+	httpResponse, err := a.client.Do(req)
+	if err != nil {
+		return Location{}, fmt.Errorf("could not execute location get-by-code request: %w", err)
+	}
+	if httpResponse.StatusCode >= 500 && httpResponse.StatusCode < 600 {
+		return Location{}, fmt.Errorf("could not execute location get-by-code request, got response %s", httpResponse.Status)
+	}
+
+	var location Location
+	err = json.NewDecoder(httpResponse.Body).Decode(&location)
+	_ = httpResponse.Body.Close()
```

Comment:
use a `defer httpResponse.Body.Close()` right after the error check for `a.client.Do()` - look at #48 for lots of examples

Reply:
Done via 346a0c3